code/game/objects/items.dm

```diff
@@ -132,6 +132,21 @@ GLOBAL_DATUM_INIT(welding_sparks_prepdoor, /mutable_appearance, mutable_appearan
 
 	var/active = FALSE
 
+
+
+	//Coloring vars
+	///Some defines to determin if the item is allowed to be recolored.
+	var/colorable_allowed = NONE
+	///optional assoc list of colors we can color this item
+	var/list/colorable_colors = list()
+	///List of icon_state suffixes for item varients.
+	var/list/icon_state_variants = list()
+	///Current varient selected.
+	var/current_variant
+
+
+
```

Comment:
Going to need to look at these indepth, seems off to me
Especially the variant var